### ISSUE TITLE
WebOS: DV files  do not play on my OLED48A2PSA

### ISSUE THREAD

**Issue by RohanKrishna500** [xbmc/xbmc#25278]
### Describe the bug
Here is a clear and concise description of what the problem is:

Any Dolby Vision Files that are in mkv format ,doesn't matter if its hybrid or single layer and any dv profiles fail to play on the latest build on kodi both the main build and the nightly build and always result in blackscreen video while audio is playing in the background.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Dolby Vision logo should pop up with fine colours and video playback without blackscreen.

## Actual Behavior

blackscreen video while audio is playing in the background.

## Possible Fix

https://www.reddit.com/r/Addons4Kodi/comments/1c25gs2/kodi_21_omega_getting_black_screen_with_some_dv/
even this isnt working.

### To Reproduce

1.  play this file https://mega.nz/file/NkhjyIhB#eCfEgW5LgGILu498Zfm_AZsRZOWyJm3R5tbe4fw-xuY

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/raw/yebuhexiqi

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: WebOS 8.3.0-2401(number1-nameri)
 - Kodi version: 21.0.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by quietvoid** [comment]
Your log is missing debug logs. Enable debug logging before playback.

**Comment by RohanKrishna500** [comment]
Alright will edit this and send it here

**Comment by RohanKrishna500** [comment]
> Your log is missing debug logs. Enable debug logging before playback.

edited and uploaded,please check now

**Comment by quietvoid** [comment]
It doesn't seem to be correctly setting Dolby Vision:
> 2024-05-30 17:33:38.287 T:4623    debug <general>: PlayerCallback: CDVDVideoCodecStarfish: type: 4, numValue: 0, strValue: {"context":"_l87JQ2VtyfAnPU","content":"movie","video":{"frameRate":23.9759998,"scanType":"VIDEO_PROGRESSIVE","width":3840,"height":1600,"pixelAspectRatio":{"width":1,"height":1},"bitRate":0,"adaptive":false,"path":"network","afd":16,"rotation":0,"data3D":{"originalPattern":"2d","currentPattern":"2d","typeLR":"LR"},"hdrType":"HDR10","mediaSei":{"displayPrimariesX0":13250,"displayPrimariesX1":7500,"displayPrimariesX2":34000,"displayPrimariesY0":34500,"displayPrimariesY1":3000,"displayPrimariesY2":16000,"whitePointX":15635,"whitePointY":16450,"minDisplayMasteringLuminance":1,"maxDisplayMasteringLuminance":10000000,"maxContentLightLevel":422,"maxPicAverageLightLevel":189},"mediaVui":{"transferCharacteristics":16,"colorPrimaries":9,"matrixCoeffs":9}}}

**Comment by RohanKrishna500** [comment]
> It doesn't seem to be correctly setting Dolby Vision:
> 
> > 2024-05-30 17:33:38.287 T:4623    debug : PlayerCallback: CDVDVideoCodecStarfish: type: 4, numValue: 0, strValue: {"context":"_l87JQ2VtyfAnPU","content":"movie","video":{"frameRate":23.9759998,"scanType":"VIDEO_PROGRESSIVE","width":3840,"height":1600,"pixelAspectRatio":{"width":1,"height":1},"bitRate":0,"adaptive":false,"path":"network","afd":16,"rotation":0,"data3D":{"originalPattern":"2d","currentPattern":"2d","typeLR":"LR"},"hdrType":"HDR10","mediaSei":{"displayPrimariesX0":13250,"displayPrimariesX1":7500,"displayPrimariesX2":34000,"displayPrimariesY0":34500,"displayPrimariesY1":3000,"displayPrimariesY2":16000,"whitePointX":15635,"whitePointY":16450,"minDisplayMasteringLuminance":1,"maxDisplayMasteringLuminance":10000000,"maxContentLightLevel":422,"maxPicAverageLightLevel":189},"mediaVui":{"transferCharacteristics":16,"colorPrimaries":9,"matrixCoeffs":9}}}

i tried all the solution like hdr format with/without DV and HDR and the video only worked when i disabled both of these but it shows HDR logo instead of DV logo

**Comment by RohanKrishna500** [comment]
> It doesn't seem to be correctly setting Dolby Vision:
> 
> > 2024-05-30 17:33:38.287 T:4623    debug : PlayerCallback: CDVDVideoCodecStarfish: type: 4, numValue: 0, strValue: {"context":"_l87JQ2VtyfAnPU","content":"movie","video":{"frameRate":23.9759998,"scanType":"VIDEO_PROGRESSIVE","width":3840,"height":1600,"pixelAspectRatio":{"width":1,"height":1},"bitRate":0,"adaptive":false,"path":"network","afd":16,"rotation":0,"data3D":{"originalPattern":"2d","currentPattern":"2d","typeLR":"LR"},"hdrType":"HDR10","mediaSei":{"displayPrimariesX0":13250,"displayPrimariesX1":7500,"displayPrimariesX2":34000,"displayPrimariesY0":34500,"displayPrimariesY1":3000,"displayPrimariesY2":16000,"whitePointX":15635,"whitePointY":16450,"minDisplayMasteringLuminance":1,"maxDisplayMasteringLuminance":10000000,"maxContentLightLevel":422,"maxPicAverageLightLevel":189},"mediaVui":{"transferCharacteristics":16,"colorPrimaries":9,"matrixCoeffs":9}}}

this issue seems to be affected to A series model so i think a Custom Build is Required for this.
Meanwhile i will upload the log when in the video settings only DV is selected as HDR format and HDR10+ disabled

**Comment by RohanKrishna500** [comment]
> It doesn't seem to be correctly setting Dolby Vision:
> 
> > 2024-05-30 17:33:38.287 T:4623    debug : PlayerCallback: CDVDVideoCodecStarfish: type: 4, numValue: 0, strValue: {"context":"_l87JQ2VtyfAnPU","content":"movie","video":{"frameRate":23.9759998,"scanType":"VIDEO_PROGRESSIVE","width":3840,"height":1600,"pixelAspectRatio":{"width":1,"height":1},"bitRate":0,"adaptive":false,"path":"network","afd":16,"rotation":0,"data3D":{"originalPattern":"2d","currentPattern":"2d","typeLR":"LR"},"hdrType":"HDR10","mediaSei":{"displayPrimariesX0":13250,"displayPrimariesX1":7500,"displayPrimariesX2":34000,"displayPrimariesY0":34500,"displayPrimariesY1":3000,"displayPrimariesY2":16000,"whitePointX":15635,"whitePointY":16450,"minDisplayMasteringLuminance":1,"maxDisplayMasteringLuminance":10000000,"maxContentLightLevel":422,"maxPicAverageLightLevel":189},"mediaVui":{"transferCharacteristics":16,"colorPrimaries":9,"matrixCoeffs":9}}}

Updated the log again while only DV selected as HDR format

**Comment by RohanKrishna500** [comment]
> It doesn't seem to be correctly setting Dolby Vision:
> 
> > 2024-05-30 17:33:38.287 T:4623    debug : PlayerCallback: CDVDVideoCodecStarfish: type: 4, numValue: 0, strValue: {"context":"_l87JQ2VtyfAnPU","content":"movie","video":{"frameRate":23.9759998,"scanType":"VIDEO_PROGRESSIVE","width":3840,"height":1600,"pixelAspectRatio":{"width":1,"height":1},"bitRate":0,"adaptive":false,"path":"network","afd":16,"rotation":0,"data3D":{"originalPattern":"2d","currentPattern":"2d","typeLR":"LR"},"hdrType":"HDR10","mediaSei":{"displayPrimariesX0":13250,"displayPrimariesX1":7500,"displayPrimariesX2":34000,"displayPrimariesY0":34500,"displayPrimariesY1":3000,"displayPrimariesY2":16000,"whitePointX":15635,"whitePointY":16450,"minDisplayMasteringLuminance":1,"maxDisplayMasteringLuminance":10000000,"maxContentLightLevel":422,"maxPicAverageLightLevel":189},"mediaVui":{"transferCharacteristics":16,"colorPrimaries":9,"matrixCoeffs":9}}}

Hi,any update on this?

**Comment by sarbes** [comment]
No. Quit nagging.

**Comment by RohanKrishna500** [comment]
> No. Quit nagging.

hi,sorry didnt get you,will it be fixed?

**Comment by sarbes** [comment]
No. If you don't have anything meaningful to contribute, refrain from posting.

**Comment by RohanKrishna500** [comment]
> No. If you don't have anything meaningful to contribute, refrain from posting.

ok understood,can u tell me any workaround or a different approach so that I can try myself.
